src/openms/include/OpenMS/ANALYSIS/QUANTITATION/IsotopeLabelingMDVs.h

```diff
@@ -0,0 +1,204 @@
+      The formula is extracted from "High-resolution 13C metabolic flux analysis",
+      Long et al, doi:10.1038/s41596-019-0204-0
+
+      @param[in]  normalized_featureMap FeatureMap with normalized values for each component and the number of heavy labeled e.g., carbons.
+      @param[in]  experiment_data Experiment data in percent.
+      @param[in]  isotopic_purity_name Name of the isotopic purity tracer to be saved as a meta value.
+      @param[out] featureMap_with_isotopic_purity FeatureMap with the calculated isotopic purity for the component group.
+    */
+    void calculateIsotopicPurities(
+      const FeatureMap& normalized_featureMap, FeatureMap& featureMap_with_isotopic_purity,
+      std::vector<double>& experiment_data, std::string& isotopic_purity_name);
+
+    /**
+      @brief This function calculates the accuracy of the MDV as compared to the theoretical MDV (only for 12C quality control experiments)
+      using average deviation to the mean.
+   
+      @param[in]  normalized_feature Feature with normalized values for each component and the chemical formula of the component group.
+      @param[in]  fragment_isotopomer_measured Measured scan values.
+      @param[in]  fragment_isotopomer_theoretical Theoretical scan values.
+      @param[out] feature_with_accuracy_info Feature with the component group accuracy and accuracy for the error for each component.
+    */
+    void calculateMDVAccuracy(
+      const Feature& normalized_feature, Feature& feature_with_accuracy_info,
+      const std::vector<double>& fragment_isotopomer_measured, const std::vector<double>& fragment_isotopomer_theoretical);
+    
+    /**
+       @brief This function calculates the accuracy of the MDVs as compared to the theoretical MDVs (only for 12C quality control experiments)
+       using average deviation to the mean.
+    
+       @param[in]  normalized_featuremap FeatureMap with normalized values for each component and the chemical formula of the component group.
+       @param[in]  fragment_isotopomer_measured Measured scan values.
+       @param[in]  fragment_isotopomer_theoretical Theoretical scan values.
+       @param[out] featuremap_with_accuracy_info FeatureMap with the component group accuracy and accuracy for the error for each component.
+    */
+    void calculateMDVAccuracies(
+      const FeatureMap& normalized_featureMap, FeatureMap& featureMap_with_accuracy_info,
+      const std::vector<double>& fragment_isotopomer_measured, const std::vector<double>& fragment_isotopomer_theoretical);
+ 
+    /**
+      @brief This function calculates the mass distribution vector (MDV)
+      either normalized to the highest mass intensity (norm_max) or normalized
+      to the sum of all mass intensities (norm_sum)
+     
+      @param[in]   mass_intensity_type Mass intensity type (either norm_max or norm_sum).
+      @param[in]   feature_name Feature name to determine which features are needed to apply calculations on.
```

Comment:
not clear from the interface what is allowed here.

I can see why 'string' is convenient (access to metavalues), but ENUM with an internal array which maps to the string you need would be much appreciated.

Reply:
thanks for the suggestion, is it recommended to have `mass_intensity_type` and `feature_name` passed in as ENUM ? for the latter there could be infinite amount of possibilities for `feature_name` beside intensity, or would it be sufficient to start with an ENUM with just 2 elements `{INTENSITY, PEAK_APEX_INT}` ?

Reply:
Exactly. There would be a final Value called `SIZE_OF...` which allows to know the number of entries in the enum.
See https://github.com/OpenMS/OpenMS/blob/develop/src/openms/include/OpenMS/METADATA/Instrument.h#L70
for an example, e.g.
```
enum IonOpticsType
{
  UNKNOWN,                  ///< unknown
  MAGNETIC_DEFLECTION,      ///< magnetic deflection
...
  SIZE_OF_IONOPTICSTYPE
};

/// Names of inlet types
static const std::string NamesOfIonOpticsType[SIZE_OF_IONOPTICSTYPE];
```

Reply:
thanks for the suggestion, it has been updated to used the enums elements.